### ISSUE TITLE
CloudBlockBlob.DownloadToStreamAsync throws StorageException on WP8.1. Works in emulator

### ISSUE THREAD

**Issue by PaulDotNet** [Azure/azure-storage-net#209]
Call to CloudBlockBlob.DownloadToStreamAsync throws StorageException with message "The remote server returned an error: Forbidden." on WP8.1. The same code works in emulator without problem.

Code is here.
```
// Declaration
        public CloudBlobContainer container;
        public CloudStorageAccount storageAccount;
        public CloudBlobClient blobClient;

// Retrieving data
            storageAccount = CloudStorageAccount.Parse(BlobAccountKey);
            blobClient = storageAccount.CreateCloudBlobClient();
            container = blobClient.GetContainerReference(containerName);

            using (Stream memoryStream = new MemoryStream())
            {
                CloudBlockBlob blockBlob = container.GetBlockBlobReference(blobName);
                await blockBlob.DownloadToStreamAsync(memoryStream);
                memoryStream.Position = 0;
                using (StreamReader streamReader = new StreamReader(memoryStream))
                {
                    return streamReader.ReadToEnd();
                }
            }
```

More exception data.

```
RequestInformation	{Microsoft.WindowsAzure.Storage.RequestResult}	Microsoft.WindowsAzure.Storage.RequestResult
ContentMd5	null	string
EgressBytes	0	long
Etag	null	string
Exception	{"The remote server returned an error: Forbidden."}	System.Exception {Microsoft.WindowsAzure.Storage.StorageException}
HttpStatusCode	403	int
HttpStatusMessage	"Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature."	string
IngressBytes	1536	long
TargetLocation	Primary	Microsoft.WindowsAzure.Storage.StorageLocation
exception	{"The remote server returned an error: Forbidden."}	System.Exception {Microsoft.WindowsAzure.Storage.StorageException}
```

Stack trace from exception object:
```
   at Microsoft.WindowsAzure.Storage.Core.Executor.Executor.EndExecuteAsync[T](IAsyncResult result)
   at Microsoft.WindowsAzure.Storage.Blob.CloudBlob.EndDownloadRangeToStream(IAsyncResult asyncResult)
   at Microsoft.WindowsAzure.Storage.Blob.CloudBlob.EndDownloadToStream(IAsyncResult asyncResult)
   at System.Action`1.Invoke(T obj)
   at Microsoft.WindowsAzure.Storage.Core.Util.AsyncExtensions.<>c__DisplayClass3_0.<CreateCallbackVoid>b__0(IAsyncResult ar)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at VFRNavigator.Blob.BlobUtils.<DownloadFileAsString>d__11.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at App.DownloadPage.<UpdateViewModel>d__10.MoveNext()
```

**Comment by pemari-msft** [comment]
This is the same as the other issue you reported. Please stick with 6.0.0 for now until we get a fix in for you. Thanks!

**Comment by PaulDotNet** [comment]
6.0.0 works but for some reason I always get "NotFound" when accessing blob the first time. The subsequent calls to retrieve the same blob are successful.

**Comment by pemari-msft** [comment]
Fixed in our 6.2 release. Please reopen and let me know if you're seeing this issue in that version of the library.
